src/views/catalog/CreateFromInstanceTypes/utils/utils.ts

```diff
@@ -31,28 +34,41 @@ const getCloudInitUserNameByOS = (selectedPreferenceName: string): string => {
 export const createPopulatedCloudInitYAML = (
   selectedPreference: string,
   subscriptionData: RHELAutomaticSubscriptionData,
+  autoUpdateEnabled?: boolean,
 ) => {
   const { activationKey, organizationID } = subscriptionData;
-  return convertUserDataObjectToYAML(
-    {
-      user: getCloudInitUserNameByOS(selectedPreference),
-      password: generateCloudInitPassword(),
-      chpasswd: { expire: false },
-      ...(selectedPreference.includes(OS_NAME_TYPES.rhel) &&
-        !isEmpty(activationKey) &&
-        !isEmpty(organizationID) && {
-          rh_subscription: { [ACTIVATION_KEY]: activationKey, org: organizationID },
-        }),
-    },
-    true,
-  );
+
+  const cloudInitConfig: CloudInitUserData = {
+    user: getCloudInitUserNameByOS(selectedPreference),
+    password: generateCloudInitPassword(),
+    chpasswd: { expire: false },
+  };
+
+  if (
+    selectedPreference.includes(OS_NAME_TYPES.rhel) &&
```

Comment:
Consider saving this to a variable with an explanatory name to make this more readable.